src/main/java/algorithms/curated170/medium/countunivaluesubtrees/CountUnivalueSubtreesDFS.java

```diff
@@ -0,0 +1,33 @@
+package algorithms.curated170.countunivaluesubtrees.medium;
+
+import algorithms.datastructures.TreeNode;
+
+public class CountUnivalueSubtreesDFS {
+
+    private int count = 0;
+
+    public int countUnivalSubtrees(TreeNode root) {
+        if (root == null) {
+            return 0;
+        }
+        isUnivalSubtree(root);
+        return count;
+    }
+
+    private boolean isUnivalSubtree(TreeNode root) {
+
+        boolean isUnival = true;
+        if (root.left != null) {
+            isUnival = isUnivalSubtree(root.left) && root.left.val == root.val;
+        }
+        if (root.left != null) {
```

Comment:
```suggestion
        if (root.right!= null) {
```
Quite fine when we fix this:
![image](https://user-images.githubusercontent.com/60903744/120088356-1f086b80-c0f8-11eb-937d-ae69da23b286.png)